src/index.ts

```diff
@@ -219,9 +355,20 @@ function cachedLookup <T> (fn: (arg: string) => T): (arg: string) => T {
  * Register TypeScript compiler instance onto node.js
  */
 export function register (opts: RegisterOptions = {}): Register {
-  const options = { ...DEFAULTS, ...opts }
+  return registerInternal(new OptionsHelper({
```

Comment:
Can we keep only a single `register` function - not clear why we need a layer of indirection. If we implement options clearer it should be resolved.